### PR TITLE
ReaderGesture: cleanup

### PR DESCRIPTION
**Pull request by yparitcher**
Remove duplicate function

More commits coming

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/6292)

### CODE REVIEW

**Comment by yparitcher** [comment]
Ready for review.

part one of a cleanup of `ReaderGesture` making all actions events and putting functions in their appropriate files.

all the network listeners were put into a new `networklistener.lua` that is registered with both the reader and FM and can always handle network events. thanks to @pazos for #6314 allowing me to test this in the emulator

all device listeners (night mode, FL, GSensor) were put into a new `devicelistener.lua` that is registered with both the reader and FM and can always handle device events. this also supersedes `ReaderFrontlight` which was only available in the reader leading to a lot of hacks, this cleans that all up

please test.

**Comment by yparitcher** [comment]
Whoops forgot to add devicelistener.lua :)

File: plugins/statistics.koplugin/main.lua [hunk 1]
```diff
@@ -2101,4 +2069,36 @@ function ReaderStatistics:getReadBookByDay(month)
     return per_day
 end
 
+function ReaderStatistics:onGetReaderProgress()
```
Comment (poire-z) [inline plugins/statistics.koplugin/main.lua:2072]: I guess you could now rename all these from `onGet...` to `onShow...`

File: frontend/device/devicelistener.lua [hunk 2]
```diff
@@ -150,43 +207,29 @@ function ReaderFrontLight:onShowOnOff()
...
 
-function ReaderFrontLight:onShowIntensity()
-    if not Device:hasFrontlight() then return true end
-    local powerd = Device:getPowerDevice()
+function DeviceListener:onShowFlDialog()
+    local FrontLightWidget = require("ui/widget/frontlightwidget")
+    UIManager:show(FrontLightWidget:new{
+        use_system_fl = Device:hasLightLevelFallback()
+    })
+end
```
Comment (Frenzie) [inline frontend/device/devicelistener.lua:215]: It does keep the diff nice and clean but before merging it's probably better to indent for future readers.

Comment (yparitcher) [reply frontend/device/devicelistener.lua]: done

**Comment by yparitcher** [comment]
Finaly finished with this.

all gestures now call events

File: frontend/apps/reader/readerui.lua [hunk 3]
```diff
@@ -369,6 +370,13 @@ function ReaderUI:init()
         view = self.view,
         ui = self,
     })
+    --  event listener for reader
+    self:registerModule("readerlistener", ReaderListener:new{
```
Comment (poire-z) [inline frontend/apps/reader/readerui.lua:374]: Is it worth adding yet another module ? Or do you think you'll need that to add other stuff ?

Comment (yparitcher) [reply frontend/apps/reader/readerui.lua:374]: I would rather not have added one but togglereadingorder does not fit into any of the existing ones. any better ideas?

Comment (poire-z) [reply frontend/apps/reader/readerui.lua:374]: Given that it's ReaderRolling and ReaderPaging that holds `self.inverse_reading_order` and handle page turning, I would just have that handled in each of them (so, duplicated, but it's consistent - and they are mutually exclusive).

Comment (Frenzie) [reply frontend/apps/reader/readerui.lua:374]: I agree that a little bit of duplication should be fine for this one, plus that's how it's been, hasn't it? (Albeit with some comment saying it shouldn't be.)

Comment (yparitcher) [reply frontend/apps/reader/readerui.lua]: fixed.
discussed earlier in #5423

**Comment by yparitcher** [comment]
any other objections?
I am ready to merge this

**Comment by poire-z** [comment]
There's a little remnant that still references the method you renamed to onUpdateFooter:
https://github.com/koreader/koreader/blob/62daa7b7839f6ecf4910939f46f1723586bb7943/frontend/apps/reader/modules/readerfooter.lua#L1794-L1796
(Dunno if it has any purpose, and if it should be removed or updated - but the current state should cause no crash as it replaces a no-op function with nil :)

**Comment by yparitcher** [comment]
> There's a little remnant that still references the method you renamed to onUpdateFooter:
> 
> https://github.com/koreader/koreader/blob/62daa7b7839f6ecf4910939f46f1723586bb7943/frontend/apps/reader/modules/readerfooter.lua#L1794-L1796
> 
> (Dunno if it has any purpose, and if it should be removed or updated - but the current state should cause no crash as it replaces a no-op function with nil :)

It ensures that the footer is updated when the zoom, fontsize etc. is changed. fixed in #6392

**Pull request merged**